src/Altinn.AccessManagement/Controllers/MaskinportenSchemaController.cs

```diff
@@ -82,15 +84,16 @@ public async Task<ActionResult<List<MPDelegationExternal>>> GetMaskinportenDeleg
             }
             catch (ArgumentException ex)
             {
-                return BadRequest(ex.Message);
+                ModelState.AddModelError(ex.ParamName ?? "Validation Error", ex.Message);
+                return new ObjectResult(ProblemDetailsFactory.CreateValidationProblemDetails(HttpContext, ModelState));
             }
             catch (Exception ex)
             {
                 _logger.LogError(ex, "GetAllDelegationsForAdmin failed to fetch delegations");
```

Comment:
Not your change, but could update the reference to the old name of the function "GetAllDelegationsForAdmin"